src/dolphin/downloadDolphin.ts

```diff
@@ -167,7 +167,12 @@ async function installDolphin(
       }
 
       log(`Extracting to: ${dolphinPath}`);
-      await extractDmg(assetPath, dolphinPath);
+      try {
+        await extractDmg(assetPath, dolphinPath);
+      } catch (err) {
+        log("Could not extract dmg");
+        console.log(err);
+      }
```

Comment:
Why would we want to catch the error here? Wouldn't the rest of the function fail if the call to `extractDmg` failed?

Reply:
weirdly, it doesn't, the rest of the code continues and doesn't overtly fail (apart from the chmod stuff but that is in a try catch already).

Honestly it might be better to just return in that catch as a fair early.

Reply:
This is `installDolphin` function is getting kind of overly complicated.

Can you instead create 3 (or 2 if you exclude windows) new functions for installingDolphin for each individual OS?

So essentially the current `installDolphin` will at the switch statement for the OS, simply call a `installDolphinOnMac()` and `installDolphinOnLinux()` function?

Then in those OS-specific install functions, you only have to handle the optimistic path, and not need to worry about catching any errors.

The parent `installDolphin` can be the one to handle the errors if necessary.

Reply:
I left a single try catch in the Mac and Linux functions because it is likely that they wouldn't complete fully if I let the caller catch them instead.